src/validation.rs

```diff
@@ -0,0 +1,94 @@
+/// .
+/// Check if input is a valid REAL
+/// NaN or infinite will return FALSE
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn IS_VALID__REAL(input: f32) -> bool {
+    !(input.is_nan() || input.is_infinite())
+}
+
+/// .
+/// Check if input is a valid LREAL
+/// NaN or infinite will return FALSE
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn IS_VALID__LREAL(input: f64) -> bool {
+    !(input.is_nan() || input.is_infinite())
+}
+
+/// .
+/// Check if input as a valid BCD
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn IS_VALID_BCD__BYTE(input: u8) -> bool {
+    let bits_per_bcd_digit = 4;
+    let iterations = u8::BITS / bits_per_bcd_digit;
+    let mut valid = true;
+    for i in 0..iterations {
+        if input >> (bits_per_bcd_digit * i) & 0b1111 > 9 {
+            valid = false;
```

Comment:
if you return false at this line, and true at the end of the function you save the extra var